onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -192,6 +266,57 @@ Status Attention<T>::UseSharedPrePackedBuffers(std::vector<BufferUniquePtr>& pre
   return Status::OK();
 }
 
+template <typename T>
+std::optional<Tensor> Attention<T>::GetPrePackTensor(int input_index) {
+  if (1 != input_index) {
+    return std::nullopt;
+  }
+
+  return std::move(packed_tensor_);
+}
+
+template <typename T>
+Status Attention<T>::SetPrePackTensor(int input_idx, const Tensor& pre_packed_tensor) {
+  if (1 != input_idx) {
+    return Status::OK();
+  }
+
+  ConvertTensorToPrePackWeight(const_cast<void*>(pre_packed_tensor.DataRaw()));
+  is_prepack_ = true;
+
+  return Status::OK();
+}
+
+template <typename T>
+void Attention<T>::ConvertTensorToPrePackWeight(void* tensor_data_raw) {
+  // buffer of packed_tensor is combine of:
```

Comment:
Can we create a struct of the layout and then reinterpret the ptr into it, it is difficult to read and understand the code.